Scripts/RemediationScripts/Remediate-DisableRemoteDebuggingForAppServices.ps1

```diff
@@ -0,0 +1,886 @@
+
+    $totalAppServices = ($validAppServiceDetails | Measure-Object).Count
+
+    if ($totalAppServices -eq 0)
+    {
+        Write-Host "No App Services found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+
+    Write-Host "Found $($totalAppServices) App Service(s)." -ForegroundColor $([Constants]::MessageType.Update)
+    
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\EnableRemoteDebuggingForAppServices"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+    
+    if (-not $Force)
+    {
+        Write-Host "Do you want to enable Remote Debugging on the production slot and all non-production slots for all App Services?" -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+            
+        $userInput = Read-Host -Prompt "(Y|N)"
+
+        if($userInput -ne "Y")
+        {
+            Write-Host "Remote Debugging will not be enabled for any App Service. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+            break
+        }
+    }
+    else
+    {
+        Write-Host "'Force' flag is provided. Remote Debugging will be enabled on the production slot and all non-production slots for all App Services without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+    }
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "`n[Step 3 of 3] Enabling Remote Debugging for App Services..."
+
+    # Includes App Services, to which, previously made changes were successfully rolled back.
+    $appServicesRolledBack = @()
+
+    # Includes App Services that were skipped during roll back. There were errors rolling back the changes made previously.
+    $appServicesSkipped = @()
+
+    $validAppServiceDetails | ForEach-Object {
+        $appService = $_
+        $resourceId = $_.ResourceId
+        $resourceGroupName = $_.ResourceGroupName
+        $resourceName = $_.ResourceName
+
+        try
+        {
+            $isRemoteDebuggingDisabledOnProductionSlot = -not $(Get-AzWebApp -ResourceGroupName $resourceGroupName -Name $resourceName -ErrorAction SilentlyContinue).SiteConfig.RemoteDebuggingEnabled
+
+            if (-not $isRemoteDebuggingDisabledOnProductionSlot)
+            {
+                Write-Host "Remote Debugging is already enabled on the production slot." -ForegroundColor $([Constants]::MessageType.Warning)
+            }
+
+            $resource = Get-AzWebApp -ResourceGroupName $resourceGroupName -Name $resourceName
```

Comment:
This Get-AzWebApp call is redundant, we have already done this at line # 699.

Reply:
Done